src/transaction/log_prior_send.hpp

```diff
@@ -46,15 +47,18 @@ namespace cublog
       prior_sender &operator = (prior_sender &&) = delete;
 
     public:
-      void send_list (const log_prior_node *head);                // send prior node list to all sinks
+      void send_list (const log_prior_node *head,
+		      const LOG_LSA *unsent_lsa);                // send prior node list to all sinks
 
-      void add_sink (const sink_hook_t &fun);                     // add a hook for a new sink
+      LOG_LSA add_sink (const sink_hook_t &fun);                     // add a hook for a new sink
       void remove_sink (const sink_hook_t &fun);                  // add a hook for a new sink
+      void reset_unsent_lsa (const LOG_LSA &lsa);                    // reset only when prior_lsa is reset
 
     private:
       // non-owning pointers
       std::vector<const sink_hook_t *> m_sink_hooks;              // hooks for sinks
       std::mutex m_sink_hooks_mutex;                              // protect access on sink hooks
+      LOG_LSA m_unsent_lsa;                                          // lsa log records to send, in other word, unsent
```

Comment:
I do not say that `0|0` is better.
I just say that - without and explicit initialization - like `LSA_INITIALIZER`, the `LOG_LSA` ctor initializes the values to `0|0` which is a valid value.
And we would want to initialize to some invalid value - `-1|-1`.

The legacy ctor of `log_lsa` is wrong and - indeed - we are by no means going to fix it now because .. legacy.

`prior_sender () : m_unsent_lsa { -1, -1}` is definitely the correct way.

I am sorry for my message. I realize now that it is confusing. I will try to be more precise.

Reply:
I do not say that `0|0` is better.
I just say that - without and explicit initialization - like `LSA_INITIALIZER`, the `LOG_LSA` ctor initializes the values to `0|0` which is a valid value.
And we would want to initialize to some invalid value - `-1|-1`.

The legacy ctor of `log_lsa` is wrong and - indeed - we are by no means going to fix it now because .. legacy.

`prior_sender () : m_unsent_lsa { -1, -1}` is definitely the correct way.

I am sorry for my message. I realize now that it is confusing. I will try to be more precise.